# HG changeset patch
# User Rik <rik@octave.org>
# Date 1540917214 25200
#      Tue Oct 30 09:33:34 2018 -0700
# Node ID 63b3d5e4e5c326782e7f767b575750ca8da03098
# Parent  221c039aa4156983f0a4cee95d5b7802e4ccfa78
Allow 'cdata' property of uicontrol to be set to '[]' (bug #54918).

* graphics.in.h (uicontrol::properties::init): Add size constraint
of "dim_vector (0,0)" to cdata property.

diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -5469,16 +5469,17 @@ public:
 
   protected:
     void init (void)
     {
       cdata.add_constraint ("double");
       cdata.add_constraint ("single");
       cdata.add_constraint ("uint8");
       cdata.add_constraint (dim_vector (-1, -1, 3));
+      cdata.add_constraint (dim_vector (0, 0));
       position.add_constraint (dim_vector (1, 4));
       sliderstep.add_constraint (dim_vector (1, 2));
       fontsize.add_constraint ("min", 0.0, false);
       cached_units = get_units ();
     }
 
     void update_text_extent (void);
 
